Add documentation for const keyword (#1024)
pyccel/pyccel: tutorial/const_keyword.md

# Const keyword

> In case if you want to make sure that an argument will not change inside its function, Pyccel provides a `const` keyword which is a qualifier of the data type of the variable and which will be converted to its equivalents in C/Fortran, Here is a simple example of its usage:

Reviewer: `In case` is not necessary, you can start the sentence with `If`

> In case if you want to make sure that an argument will not change inside its function, Pyccel provides a `const` keyword which is a qualifier of the data type of the variable and which will be converted to its equivalents in C/Fortran, Here is a simple example of its usage:

Reviewer: I suggest using `the target language` instead of `C/Fortran`. It means one less thing to change if/when we add an additional language

> If you want to make sure that an argument will not change inside its function, Pyccel provides a `const` keyword which is a qualifier of the data type of the variable and which will be converted to its equivalents in the target language, Here is a simple example of its usage:

Reviewer 2: I suggest some improvements:
```suggestion
In order to make sure that a function argument is not modified by the function call, Pyccel provides the `const` keyword, which is converted to an equivalent datatype qualifier in the target language. Here is a simple example of its usage:
```

Here is a working Python code with `const` keyword:

```python
def func1(arr: 'const int[:]'):
    # some code
    return 0
```

The generated C code:

```C
#include "boo.h"
#include "ndarrays.h"
#include <stdint.h>
#include <stdlib.h>

/*........................................*/
int64_t func1(t_ndarray arr)
{
    /*some code*/
    return 0;
}
/*........................................*/
```

> The fortran equevalent:

Reviewer 2: Typo: equevalent -> equivalent
